airflow/providers/amazon/aws/operators/glue_session.py

```diff
@@ -0,0 +1,260 @@
+
+    :param session_id: session id
+    :param session_desc: session description
+    :param region_name: aws region name (example: us-east-1)
+    :param iam_role_name: AWS IAM Role for Glue Session Execution. If set `iam_role_arn` must equal None.
+    :param iam_role_arn: AWS IAM Role ARN for Glue Session Execution, If set `iam_role_name` must equal None.
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Session
+    :param create_session_kwargs: Extra arguments for Glue Session Creation
+    """
+
+    template_fields: Sequence[str] = (
+        "session_id",
+        "create_session_kwargs",
+        "iam_role_name",
+        "iam_role_arn",
+    )
+    template_fields_renderers = {
+        "create_session_kwargs": "json",
+    }
+
+    def __init__(
+        self,
+        *,
+        session_id: str = "aws_glue_default_session",
+        session_desc: str = "AWS Glue Session with Airflow",
+        aws_conn_id: str = "aws_default",
+        iam_role_name: str | None = None,
+        iam_role_arn: str | None = None,
+        num_of_dpus: int | None = None,
+        create_session_kwargs: dict | None = None,
+        wait_for_readiness: bool = True,
+        waiter_delay: int = 15,
+        waiter_max_attempts: int = 60,
+        deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
+        session_poll_interval: int | float = 6,
+        delete_session_on_kill: bool = False,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.session_id = session_id
+        self.session_desc = session_desc
+        self.aws_conn_id = aws_conn_id
+        self.iam_role_name = iam_role_name
+        self.iam_role_arn = iam_role_arn
+        self.create_session_kwargs = create_session_kwargs or {}
+        self.wait_for_readiness = wait_for_readiness
+        self.waiter_delay = waiter_delay
+        self.waiter_max_attempts = waiter_max_attempts
+        self.session_poll_interval = session_poll_interval
+        self.delete_session_on_kill = delete_session_on_kill
+        self.deferrable = deferrable
+
+        worker_type_exists = "WorkerType" in self.create_session_kwargs
+        num_workers_exists = "NumberOfWorkers" in self.create_session_kwargs
+
+        if self.iam_role_arn and self.iam_role_name:
+            raise ValueError("Cannot set iam_role_arn and iam_role_name simultaneously")
+        if worker_type_exists and num_workers_exists:
+            if num_of_dpus is not None:
+                raise ValueError("Cannot specify num_of_dpus with custom WorkerType")
+        elif not worker_type_exists and num_workers_exists:
+            raise ValueError("Need to specify custom WorkerType when specifying NumberOfWorkers")
+        elif worker_type_exists and not num_workers_exists:
+            raise ValueError("Need to specify NumberOfWorkers when specifying custom WorkerType")
+        elif num_of_dpus is None:
+            self.num_of_dpus: int | float = 10
```

Comment:
Is this a common default? If not it might be work mentioning it in the :param: doc string for num_of_dpus

Reply:
This is the default value of the resource in aws